python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+    resultsOutput = namedtuple('resultsOutput',
+                               ['psfFitPosCentroidX', 'psfFitPosCentroidY',
+                                'psfFitNegCentroidX', 'psfFitNegCentroidY', 'psfFitPosFlux',
+                                'psfFitNegFlux', 'psfFitPosFluxSigma', 'psfFitNegFluxSigma',
+                                'psfFitCentroidX', 'psfFitCentroidY', 'psfFitOrientation',
+                                'psfFitSignaltoNoise', 'psfFitChi2', 'psfFitRedChi2'])
+
+    def __init__(self, diffim, posImage=None, negImage=None):
+        """Algorithm to run dipole measurement on a diaSource
+        @param diffim Exposure on which the diaSources were detected
+        @param posImage "Positive" exposure from which the template was subtracted
+        @param negImage "Negative" exposure which was subtracted from the posImage
+        """
+
+        self.diffim = diffim
+        self.posImage = posImage
+        self.negImage = negImage
+        self.psfSigma = None
+        if diffim is not None:
+            self.psfSigma = diffim.getPsf().computeShape().getDeterminantRadius()
+        self.log = pexLog.Log(pexLog.Log.getDefaultLog(),
+                              'lsst.ip.diffim.DipoleFitAlgorithm', pexLog.Log.INFO)
+
+        import lsstDebug
+        self.debug = lsstDebug.Info(__name__).debug
+
+    def genBgGradientModel(self, in_x, pars=()):
+        """Generate gradient model (2-d array) with up to 2nd-order polynomial
+        @param in_x Provides the input x,y grid upon which to compute the gradient
+        @param pars Up to 6 floats for up to 6 2nd-order 2-d polynomial gradient parameters.
+        If emtpy, do nothing (for speed).
+
+        @return None, or 2-d numpy.array of width/height matching
+        input bbox, containing computed gradient values.
+        """
+
+        gradient = None
+
+        # Don't fit for other gradient parameters if the intercept is not allowed.
+        if len(pars) > 0 and pars[0] is not None:
+            y, x = in_x[0, :], in_x[1, :]
+            gradient = np.full_like(x, pars[0], dtype='float64')
+            if len(pars) > 1 and pars[1] is not None:
+                gradient += pars[1] * x
+            if len(pars) > 2 and pars[2] is not None:
+                gradient += pars[2] * y
+            if len(pars) > 3 and pars[3] is not None:
+                gradient += pars[3] * (x * y)
+            if len(pars) > 4 and pars[4] is not None:
+                gradient += pars[4] * (x * x)
+            if len(pars) > 5 and pars[5] is not None:
+                gradient += pars[5] * (y * y)
+        return gradient
+
+    def _generateXYGrid(self, bbox):
+        x, y = np.mgrid[bbox.getBeginY():bbox.getEndY(), bbox.getBeginX():bbox.getEndX()]
+        in_x = np.array([y, x]).astype(np.float64)
+        in_x[0, :] -= np.mean(in_x[0, :])
+        in_x[1, :] -= np.mean(in_x[1, :])
+        return in_x
+
+    def fitBackgroundGradient(self, source, posImage, order=1):
```

Comment:
All suggested changes made. Changed `genBgGradientModel` to 'genBackgroundModel` and `fitBackgroundGradient` to `fitBackgroundGradient` to `fitFootprintBackground`. I agree it is confusing, however, they are quite distinct tasks, and the names now accurately describe the tasks that each performs.